corehq/messaging/fcm_util.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+
+from django.conf import settings
+from firebase_admin import messaging, credentials, initialize_app
+
+logger = logging.getLogger(__name__)
+
+MAX_DEVICES_ALLOWED_MULTICAST = 500
+HQ_FCM_UTIL = None
+
+
+def is_fcm_available():
+    if not settings.FCM_CREDS_PATH:
+        return False
+    return os.path.isfile(settings.FCM_CREDS_PATH)
+
+
+class FCMUtilException(Exception):
+    pass
+
+
+class DevicesLimitExceeded(FCMUtilException):
+    def __init__(self):
+        super().__init__(f"Max devices allowed is {MAX_DEVICES_ALLOWED_MULTICAST}! Please execute in batches.")
+
+
+class FCMUtil:
+    def __init__(self, app_name, creds_path=settings.FCM_CREDS_PATH):
+        creds = credentials.Certificate(creds_path)
+        self.app = initialize_app(credential=creds, name=app_name)
+
+    @staticmethod
+    def _build_notification(title, body):
```

Comment:
nit: I don't see a purpose for this method. Why do we have this?

Reply:
This is used to create `messaging.Notification` with title and body and is reused in both send methods.

Reply:
Got it. I think as I have moved the check to this function , we can now keep this function.